portfolio/src/test/java/com/google/sps/ShortestTaskFirstTest.java

```diff
@@ -0,0 +1,187 @@
+
+    Task task1 = new Task("Task 1", "First task", TIME_20_MINUTES, LOW_PRIORITY);
+    Task task2 = new Task("Task 2", "Second task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1, task2);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0900);
+    ScheduledTask scheduledTask2 = new ScheduledTask(task2, TIME_1000);
+
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1, scheduledTask2);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that no errors are thrown for events fully outside working
+  // hours.
+  @Test
+  public void eventTotallyOutsideWorkingHours() {
+    ShortestTaskFirst algorithm = new ShortestTaskFirst();
+
+    Collection<CalendarEvent> events =
+        Arrays.asList(new CalendarEvent("Event 1", TIME_1800, TIME_2000));
+    Task task1 = new Task("Task 1", "First task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0900);
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that no errors are thrown for events partially outside working
+  // hours.
+  @Test
+  public void eventPartiallyOutsideWorkingHours() {
+    ShortestTaskFirst algorithm = new ShortestTaskFirst();
+
+    Collection<CalendarEvent> events =
+        Arrays.asList(new CalendarEvent("Event 1", TIME_0830, TIME_0930));
+    Task task1 = new Task("Task 1", "First task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0930);
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that no errors are thrown if working hours is longer than a
+  // 24 hour period.
+  @Test
+  public void workingPeriodExceedsTwentyFourHours() {
+    ShortestTaskFirst algorithm = new ShortestTaskFirst();
+
+    Collection<CalendarEvent> events = Arrays.asList();
+    Task task1 = new Task("Task 1", "First task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0900);
+
+    Collection<ScheduledTask> actual =
+        algorithm.schedule(events, tasks, TIME_0900, THREE_DAYS_LATER);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure no errors are thrown if we have more tasks than we can possibly
```

Comment:
You can consider including this info in the method name.

ex:
tooManyTasksToSchedule_shouldNotThrowError() {

Reply:
Done.